src/Tickets/Commerce/Gateways/Stripe/Hooks.php

```diff
@@ -328,6 +339,122 @@ public function maybe_create_stripe_payment_intent() {
 		tribe( Payment_Intent_Handler::class )->create_payment_intent_for_cart();
 	}
 
+	/**
+	 * Handles the checkout request pieces related to Stripe Gateway.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function handle_checkout_request() {
+		$payment_intent_id            = tec_get_request_var( 'payment_intent' );
+		$payment_intent_client_secret = tec_get_request_var( 'payment_intent_client_secret' );
+
+		if ( ! $payment_intent_id || ! $payment_intent_client_secret ) {
+			return;
+		}
+
+		$existing_payment_intent = tribe( Payment_Intent_Handler::class )->get();
+
+		// Do we need to re-fecth the payment intent?
+		if ( ! empty( $existing_payment_intent ) && ( $existing_payment_intent['id'] === $payment_intent_id || $existing_payment_intent['client_secret'] === $payment_intent_client_secret ) ) {
+			$payment_intent = $existing_payment_intent;
+		} else {
+			$payment_intent = Payment_Intent::get( $payment_intent_id );
+		}
+
+		// Invalid payment intent, bail.
+		if ( $payment_intent['client_secret'] !== $payment_intent_client_secret ) {
+			return;
+		}
+
+		// Overwrite the local payment intent, since we confirmed the one we received is the one in use.
+		// This will be relevant for the checkout page.
+		tribe( Payment_Intent_Handler::class )->set( $payment_intent );
+
+		$success_url = add_query_arg( [ 'tc-order-id' => $payment_intent['id'] ], tribe( Success::class )->get_url() );
+		$new_status  = tribe( Status::class )->convert_payment_intent_to_commerce_status( $payment_intent );
+
+		$order = tec_tc_orders()->by_args(
+			[
+				'status'           => 'any',
+				'gateway_order_id' => $payment_intent['id'],
+			]
+		)->first();
+
+		// We will attempt to update the order status to the one returned by Stripe.
+		$updated = tribe( Order::class )->modify_status(
+			$order->ID,
+			$new_status->get_slug(),
+			[
+				'gateway_payload'  => $payment_intent,
+				'gateway_order_id' => $payment_intent['id'],
+			]
+		);
+
+		// If we get a success status, we redirect to the success page.
+		if ( Completed::SLUG === $new_status->get_slug() ) {
+			wp_safe_redirect( $success_url );
+			exit;
+		}
```

Comment:
This can be problematic, for async payment methods Stripe will send `pending` status which will later be `completed` by webhooks. But for that case the user will be stuck on the checkout page and continue to create pending orders.